modules/balana-utils/src/main/java/org/wso2/balana/utils/Utils.java

```diff
@@ -60,7 +60,7 @@ public static String getStringFromDocument(Document doc) throws TransformerExcep
         DOMSource domSource = new DOMSource(doc);
         StringWriter writer = new StringWriter();
         StreamResult result = new StreamResult(writer);
-        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        TransformerFactory transformerFactory = TransformerFactory.newInstance("org.apache.xalan.processor.TransformerFactoryImpl",null);
```

Comment:
I think we better get the processor name from the system property or somewhere, and default to the previous newInstance. The apache library may not be the optimal one needed for some other use cases.
WDYT?

Reply:
Well, this is not about apache library is optimal or not. Some usage do not have apache library on the classpath due to many reasons. Hence we better not hard-fix the apache library. What I am proposing is something like,
```
String transformerFactoryClassName = System.getProperty("org.wso2.balana.TransformerFactory");
if(transformerFactoryClassName = null) {
    transformerFactoryClassName = "org.apache.xalan.processor.TransformerFactoryImpl";
}
TransformerFactory transformerFactory = TransformerFactory.newInstance(transformerFactoryClassName, null);
```

Reply:
@ruwanta 
Thank you for your suggestion. I'll incorporate these changes once I test the application thoroughly.
**Query**: When the factory class name is fetched from the System property "org.wso2.balana.TransformerFactory", in what all scenarios will the value come as not null?

Reply:
@sanchit31 
lets say some user opt for another  TransformerFactory, then they can add  something like below in the JVM args. The code will default to apache if the JVM option is not present.
```
-Dorg.wso2.balana.TransformerFactory=some.preffred.TransformerFactory
```

This is not optimal solution, but it at least allow someone to choose other transformer factory then apache.

Reply:
@ruwanta I've implemented the changes and committed it. LGTM